### DISCUSSION TITLE
Example trashed/deleted_at TernaryFilter only works if I include withTrashed() in the original query

### DISCUSSION THREAD

**Discussion by hotsaucejake** [filamentphp/filament#6753] [Help] [answered]
I'm using a table with filters, specifically the example in the documentation.  

```php
protected function getTableFilters(): array
{
        return [
            TernaryFilter::make('trashed')
                ->nullable()
                ->attribute('deleted_at')
                ->placeholder('Without Trashed')
                ->trueLabel('With Trashed')
                ->falseLabel('Only Trashed')
                ->queries(
                    true: fn (Builder $query) => $query->withTrashed(),
                    false: fn (Builder $query) => $query->onlyTrashed(),
                    blank: fn (Builder $query) => $query->withoutTrashed(),
                ),
        ];
}
```

However, the filter doesn't work unless my original query includes the trashed items:

```php
protected function getTableQuery(): Builder|Relation
{
        return auth()->user()->posts()->withTrashed()->getQuery();
}
```

Of course, the query works as I tested it in tinker, but the final query when the default page/filter loads is

```
auth()->user()->posts()->withTrashed()->withoutTrashed()->get()
```

That gives the intended result but just want to make sure everything is setup correctly.

**Comment by danharrin** [comment, marked as answer]
Hey, try replacing the entire `TernaryFilter` with `TrashedFilter`. It should all work out of the box without any methods. To be able to apply stuff to the base query, you need a custom filter class ATM.

**Reply by hotsaucejake** [reply]
Awesome, that works without the added `withTrashed()` query in the original `getTableQuery()`

It now looks like:

```php
protected function getTableQuery(): Builder|Relation
    {
        return auth()->user()->posts()->getQuery();
    }
``` 

and 

```php
protected function getTableFilters(): array
    {
        return [

            TrashedFilter::make('trashed'),
        ];
    }
```

Beautiful!

**Reply by hotsaucejake** [reply]
Further inspection, it seems the magic lies in the `applyToBaseQuery` in the TrashedFilter, good deal.

**Reply by danharrin** [reply]
Even more beautiful - `TrashedFilter::make()`

**Reply by danharrin** [reply]
Just added a `baseQuery()` to v3, so at some point you'll be able to do this without extending the class.
